### ISSUE TITLE
Poor performance (perhaps, related to adding an extension)

### ISSUE THREAD

**Issue by kachkaev** [streetsidesoftware/vscode-spell-checker#180]
Hi Jason,

I started using VSCode and the spell checker last week and noticed that the lightbulb and the `cmd+.` context menu may be slow. It sometimes takes over 10 seconds for the context menu to appear and this may even happen after I've already focused on another window. When I press `cmd+.` more than once in anticipation for the dropdown, multiple dropdowns consequently appear after a delay, which gets pretty distracting.

The same performance issue applies to opening _Spell Checker Info_. Overall, it feels like some background process simply hangs for 10-30 seconds.

Here's how my current user config looks like:
```js
    "cSpell.checkLimit": 500,
    "cSpell.language": "en_gb,en_us,ru",
    "cSpell.maxNumberOfProblems": 300,
    "cSpell.minWordLength": 3,
    "cSpell.userWords": [
        // ~ 50 words, half in EN half in RU
     ]
```

When I unset `cSpell.language` and thus fallback to `en-us`, the performance generally improves, so the issue may be related to having multiple languages working in parallel. However, it's not a 100% correlation, because sometimes things work instantly even with all three languages are enabled. Could the issue be related to memory consumption, CPU usage, garbage collection, etc.?

I'll be happy to investigate this further if you help me with instructions. My current env is the following:
```
VSCode 1.20.0
Code Spell Checker 1.4.3
Russian - Code Spell Checker 0.2.2
macOS High Sierra (latest macbook pro)
```

In any case, thanks a lot for this extension! Even with this slow performance issue, the UX is way better than in Atom where I'm coming from!

**Comment by Jason3S** [comment]
Thank you. I will take a look. 20-30 seconds is too long.

The Russian dictionary is very very large which might be adding to the issue.

**Comment by Jason3S** [comment]
Thank you for creating this issue. It is getting close now due to no activity. If you are still seeing a performance issue, please open a new issue and refer to this one.
